modules/apps/headless/headless-admin-list-type/headless-admin-list-type-impl/src/main/java/com/liferay/headless/admin/list/type/internal/resource/v1_0/ListTypeEntryResourceImpl.java

```diff
@@ -169,6 +180,24 @@ public ListTypeEntry putListTypeEntry(
 					listTypeEntry.getName_i18n())));
 	}
 
+	@Override
+	public ListTypeEntry putListTypeEntryByExternalReferenceCode(
+			String externalReferenceCode, ListTypeEntry listTypeEntry)
+		throws Exception {
+
+		com.liferay.list.type.model.ListTypeEntry serviceBuilderlistTypeEntry =
+			_listTypeEntryService.getListTypeEntryByExternalReferenceCode(
+				externalReferenceCode, contextCompany.getCompanyId());
+
+		return ListTypeEntryUtil.toListTypeEntry(
```

Comment:
To avoid duplication of code, what do you think to call ListTypeEntryResourceImpl#putListTypeEntry here like the pattern used on ObjectDefinitionResourceImpl#putObjectDefinitionByExternalReferenceCode?

Reply:
Sure!
Thanks for the tip!